StackExchange/Stacks
# feat(badge): add ai variant

[STACKS-686](https://stackoverflow.atlassian.net/browse/STACKS-686)

This PR adds `AI` and `bot` variants to the badge component. See also [Figma](https://www.figma.com/design/LG7JXmJi8gETklP1MGRkwJ/Answers-with-AI?node-id=2579-17719&t=sniAVFbOqXnzevkC-1).

## Testing

- Pull down this branch `STACKS-686/ai-badges`
- Run with `npm run start`
- Go to http://localhost:8080/product/components/badges/#user-badges
- Check out the new `AI` and `bot` badges

<details>
<summary>Screenshots</summary>

## Light

![image](https://github.com/user-attachments/assets/0f2fa45a-2c0f-4352-bad4-2cbc514dbb93)

## Light high contrast

![image](https://github.com/user-attachments/assets/7e49a1b1-33f9-43eb-8f12-5c8b5c2261bc)

## Dark

![image](https://github.com/user-attachments/assets/19239c9e-b034-49f0-819a-58e42a656f87)

## Dark high contrast

![image](https://github.com/user-attachments/assets/ec2cf89f-02c2-433d-920d-333884df89a1)

</details>

---

**Original PR description from Nov 6, 2024**: Since the request for an AI badge seems urgent, I went ahead and implemented it in this PR so we could quickly merge and ship if we decide this is the correct course. If we decide against including this in Stacks, I'm glad to junk this PR and lift the useful parts out into Core (or wherever they're needed).
